src/frontends/onnx/frontend/src/op/com.microsoft/range.cpp

```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "core/operator_set.hpp"
+#include "openvino/frontend/exception.hpp"
+#include "openvino/op/ops.hpp"
+
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace onnx {
+namespace com_microsoft {
+namespace opset_1 {
+ov::OutputVector range(const ov::frontend::onnx::Node& node) {
+    auto nodes = node.get_ov_inputs();
+    FRONT_END_GENERAL_CHECK(nodes.size() >= 2 && nodes.size() <= 3,
+                            "Range takes 2 or 3 inputs. Provided " + std::to_string(nodes.size()));
+    auto start = nodes.at(0);
+    auto limit = nodes.at(1);
+    auto step =
```

Comment:
https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#com.microsoft.Range

Documentation requires same input tensors for all inputs. Could you add a check for element types?

Like here: https://github.com/openvinotoolkit/openvino/blob/0db7f8e5d603e5212bf0731e5a08147b1e269f5b/src/frontends/onnx/frontend/src/op/com.microsoft/simplified_layer_normalization.cpp#L34

Reply:
I have added this , but this is causing repeated code which I think is not preferred , so should I put this in a loop or something like that?

Reply:
also I have changed step to delta to keep it consistent with documentation.
thank you